File: isilibrary/src/main/java/com/isi/isilibrary/products/ManageCategoryElementActivity.java
package com.isi.isilibrary.products;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.google.gson.Gson;
import com.isi.isiapi.classes.CategoryAndProduct;
import com.isi.isilibrary.IsiAppActivity;
import com.isi.isilibrary.R;
import com.isi.isilibrary.backActivity.BackActivity;

import java.util.List;

public class ManageCategoryElementActivity extends BackActivity {

    private LinearLayout linearLayout;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_manage_category_element);

        setTitle("Le tue categorie elementi");
    }

    @Override
    protected void onResume() {
        super.onResume();
        updateUI(R.layout.activity_manage_category_element);
    }

    @Override
    public void updateUI(int layout){

        super.updateUI(layout);

        linearLayout = findViewById(R.id.categoryElementLayout);
        linearLayout.removeAllViews();

        new Thread(() -> {
            List<CategoryAndProduct> categories = IsiAppActivity.isiCashierRequest.getCategories();

            runOnUiThread(() -> {
                if(categories != null){
                    for (CategoryAndProduct categories1 : categories){

                        LayoutInflater inflater = (LayoutInflater) getSystemService(LAYOUT_INFLATER_SERVICE);
                        final View inflate = inflater.inflate(R.layout.category_table, linearLayout, false);
                        TextView loadinf = inflate.findViewById(R.id.categoryTableText);
                        loadinf.setText(categories1.category.name);

                        CheckBox active = inflate.findViewById(R.id.checkbox_active);
                        CheckBox guest = inflate.findViewById(R.id.chekcbox_guest_active);

                        active.setChecked(categories1.category.active == 1);
                        guest.setChecked(categories1.category.guest == 1);

                        active.setOnCheckedChangeListener((compoundButton, b) -> {
                            categories1.category.active = b ? 1 : 0;

                            new Thread(() -> IsiAppActivity.isiCashierRequest.editcategory(categories1.category)).start();


                        });

                        guest.setOnCheckedChangeListener((compoundButton, b) -> {
                            categories1.category.guest = b ? 1 : 0;

                            new Thread(() -> IsiAppActivity.isiCashierRequest.editcategory(categories1.category)).start();

                        });

                        Button edit = inflate.findViewById(R.id.editCategoryButton);
                        edit.setOnClickListener(v -> {
                            Intent i = new Intent(ManageCategoryElementActivity.this, AddCategoryElementActivity.class);
                            i.putExtra("category", new Gson().toJson(categories1.category));
                            startActivity(i);
                        });


                        linearLayout.addView(inflate);

                    }
                }else{
                    errorPage(layout);
                }
            });

        }).start();

    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();

        inflater.inflate(R.menu.intestazione_menu, menu);

        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.addIntestazioneMenu) {
            Intent addCategory = new Intent(ManageCategoryElementActivity.this, AddCategoryElementActivity.class);
            startActivity(addCategory);
        }

        super.onOptionsItemSelected(item);

        return true;
    }
}

File: isilibrary/src/main/java/com/isi/isilibrary/products/ManageCategoryElementActivity.kt
package com.isi.isilibrary.products

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.Menu
import android.view.MenuItem
import android.widget.Button
import android.widget.CheckBox
import android.widget.CompoundButton
import android.widget.LinearLayout
import android.widget.TextView
import com.google.gson.Gson
import com.isi.isilibrary.IsiAppActivity
import com.isi.isilibrary.R
import com.isi.isilibrary.backActivity.BackActivity
import com.isi.isilibrary.dialog.NetConnection

class ManageCategoryElementActivity : BackActivity() {
    private lateinit var linearLayout: LinearLayout
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_manage_category_element)
        title = "Le tue categorie elementi"
    }

    override fun onResume() {
        super.onResume()
        updateUI(R.layout.activity_manage_category_element)
    }

    override fun updateUI(layout: Int) {
        super.updateUI(layout)
        linearLayout = findViewById(R.id.categoryElementLayout)
        linearLayout.removeAllViews()

        NetConnection(
                this,
                "Scarico categorie prodotti...",
                startNetConnection = {
                    IsiAppActivity.httpRequest?.categories
                },
                onConnectionOk = {
                    it!!.categories.sortBy { it1 -> it1.name?.lowercase() }
                    for (categories1 in it.categories) {
                        val inflater = getSystemService(LAYOUT_INFLATER_SERVICE) as LayoutInflater
                        val inflate = inflater.inflate(R.layout.category_table, linearLayout, false)

                        val loadinf = inflate.findViewById<TextView>(R.id.categoryTableText)
                        loadinf.text = categories1.name

                        val active = inflate.findViewById<CheckBox>(R.id.checkbox_active)
                        val guest = inflate.findViewById<CheckBox>(R.id.chekcbox_guest_active)

                        active.isChecked = categories1.active == 1
                        guest.isChecked = categories1.guest == 1

                        active.setOnCheckedChangeListener { _: CompoundButton?, b: Boolean ->
                            categories1.active = if (b) 1 else 0

                            Thread {
                                IsiAppActivity.httpRequest!!.editcategory(
                                        categories1
                                )
                            }
                                    .start()
                        }
                        guest.setOnCheckedChangeListener { _: CompoundButton?, b: Boolean ->
                            categories1.guest = if (b) 1 else 0

                            Thread {
                                IsiAppActivity.httpRequest!!.editcategory(
                                        categories1
                                )
                            }
                                    .start()
                        }

                        val edit = inflate.findViewById<Button>(R.id.editCategoryButton)
                        edit.setOnClickListener {
                            val i = Intent(
                                    this@ManageCategoryElementActivity,
                                    AddCategoryElementActivity::class.java
                            )
                            i.putExtra("category", Gson().toJson(categories1))
                            startActivity(i)
                        }

                        linearLayout.addView(inflate)
                    }
                },
                onConnectionError = {
                    errorPage(layout)
                }
        ).start()
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.intestazione_menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.addIntestazioneMenu) {
            val addCategory =
                Intent(this@ManageCategoryElementActivity, AddCategoryElementActivity::class.java)
            startActivity(addCategory)
        }
        super.onOptionsItemSelected(item)
        return true
    }
}